variables.tf

```
@@ -120,213 +120,213 @@ variable "create_cos_bucket" {
+  description = "The storage class of the bucket. Required only if `create_cos_bucket` is set to `true`. Possible values are `standard`, `vault`, `cold`, `smart`, or `onerate_active`."
   default     = "standard"
 
   validation {
     condition     = can(regex("^standard$|^vault$|^cold$|^smart$|^onerate_active", var.bucket_storage_class))
-    error_message = "Variable 'bucket_storage_class' must be 'standard', 'vault', 'cold', 'smart' or 'onerate_active'."
+    error_message = " The value isn't valid. Possible values are 'standard', 'vault', 'cold', 'smart', or 'onerate_active'."
   }
 }
 
 variable "management_endpoint_type_for_bucket" {
-  description = "The type of endpoint for the IBM terraform provider to manage the bucket. Possible values: `public`, `private`, `direct`."
+  description = "The type of endpoint for the IBM terraform provider to manage the bucket. Possible values are `public`, `private`, or `direct`."
   type        = string
   default     = "public"
   validation {
     condition     = contains(["public", "private", "direct"], var.management_endpoint_type_for_bucket)
-    error_message = "The specified management_endpoint_type_for_bucket is not a valid selection!"
+    error_message = "The value isn't valid. Possible values are `public`, `private`, or `direct`."
   }
 }
 
 # Where is retention (immuatble object storage) supported
 # https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-service-availability#service-availability
 variable "retention_enabled" {
-  description = "Whether retention for the Object Storage bucket is enabled. Applies only if `create_cos_bucket` is true."
+  description = "Whether retention for the Object Storage bucket is enabled. Required only if `create_cos_bucket` is set to `true`."
   type        = bool
   default     = false
   validation {
     condition     = var.cross_region_location == null || (var.cross_region_location == "us" || !var.retention_enabled)
-    error_message = "Retention is currently only supported in the `US` location for cross region buckets."
+    error_message = "Retention is currently only supported in the `US` location for cross-region buckets."
   }
 }
 
 variable "retention_default" {
-  description = "The number of days that an object can remain unmodified in an Object Storage bucket. Applies only if `create_cos_bucket` is true."
+  description = "The number of days that an object can remain unmodified in an Object Storage bucket. Required only if `create_cos_bucket` is set to `true`."
   type        = number
   default     = 90
   validation {
     condition     = var.retention_default == null ? true : (var.retention_default >= 0 && var.retention_default < 365243)
-    error_message = "The specified duration for retention maximum period is not a valid selection!"
+    error_message = "The value isn't valid. Specify any number."
```

Is this accurate for possible values - any number?

---

As per the [docs](https://cloud.ibm.com/docs/cloud-object-storage?topic=cloud-object-storage-immutable#immutable-sdk-add-policy), value should  be between `0 to 365243`, inclusive. 

```suggestion
    error_message = "The value is not valid. Enter a value from 0 to 365243, including both limits."
```